In [0]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import ttest_1samp, ttest_ind, mannwhitneyu, levene, shapiro
from statsmodels.stats.power import ttest_power
from scipy import stats 

In [12]:
from google.colab import drive
drive.mount("/content/drive")

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [0]:
df = pd.read_csv("/content/drive/My Drive/Colab Notebooks/Project-2.csv")

In [16]:
df.head()

Sales Person  Old Scheme (in Thousands)  New Scheme (in Thousands)
0             1                         57                         62
1             2                        103                        122
2             3                         59                         54
3             4                         75                         82
4             5                         84                         84

In [0]:
df.set_index('Sales Person',inplace=True)

In [18]:
df

Old Scheme (in Thousands)  New Scheme (in Thousands)
Sales Person                                                      
1                                    57                         62
2                                   103                        122
3                                    59                         54
4                                    75                         82
5                                    84                         84
6                                    73                         86
7                                    35                         32
8                                   110                        104
9                                    44                         36
10                                   82                        107
11                                   67                         84
12                                   64                         85
13                                   78                         99
14                                   53                         39
15                                   41                         34
16                                   39                         58
17                                   80                         73
18                                   87                         53
19                                   73                         66
20                                   65                         78
21                                   28                         41
22                                   62                         71
23                                   49                         38
24                                   84                         95
25                                   63                         81
26                                   77                         58
27                                   67                         75
28                                  101                         94
29                                   91                        100
30                                   50                         68

In [19]:
df.describe()

Old Scheme (in Thousands)  New Scheme (in Thousands)
count                  30.000000                  30.000000
mean                   68.033333                  71.966667
std                    20.455980                  24.162501
min                    28.000000                  32.000000
25%                    54.000000                  55.000000
50%                    67.000000                  74.000000
75%                    81.500000                  85.750000
max                   110.000000                 122.000000

In [22]:
#Seperating the two datasets 
df1 = df['Old Scheme (in Thousands)']*1000
df1

Sales Person
1      57000
2     103000
3      59000
4      75000
5      84000
6      73000
7      35000
8     110000
9      44000
10     82000
11     67000
12     64000
13     78000
14     53000
15     41000
16     39000
17     80000
18     87000
19     73000
20     65000
21     28000
22     62000
23     49000
24     84000
25     63000
26     77000
27     67000
28    101000
29     91000
30     50000
Name: Old Scheme (in Thousands), dtype: int64

In [23]:
Mean1 = df1.mean()
print("Mean of Old Scheme is %d thousand"%(Mean1))

Mean of Old Scheme is 68033 thousand


In [24]:
df2 = df['New Scheme (in Thousands)']*1000
df2

Sales Person
1      62000
2     122000
3      54000
4      82000
5      84000
6      86000
7      32000
8     104000
9      36000
10    107000
11     84000
12     85000
13     99000
14     39000
15     34000
16     58000
17     73000
18     53000
19     66000
20     78000
21     41000
22     71000
23     38000
24     95000
25     81000
26     58000
27     75000
28     94000
29    100000
30     68000
Name: New Scheme (in Thousands), dtype: int64

In [25]:
Mean2 = df2.mean()
print("Mean of Old Scheme is %d thousand"%(Mean2))

Mean of Old Scheme is 71966 thousand


**Answer-2**
> Since population statndard deviation is unknown so we have to use t stat test

> Given significance level i.e. alpha is 0.05

> Sample size = 30

**Hythothesis Formulation**

> H0 (Null hypothesis) : New scheme has not significantly raised output i.e. there is no change in means M1==M2

>H1 (Aleternate hypothesis) : New scheme has significantly raised output i.e. there is change in means M1<M2 or M2-M1>0

>Since same set of sales person sample is used and population statndard deviation is unknown so we paired sample t-test


In [29]:
DiffMean = int(Mean2-Mean1)
print('Mean difference is %d'%(DiffMean))
t_statistic,p_value = ttest_1samp(df2-df1,0)
print("t_statistic is %f , p_value is %f"%(t_statistic,p_value))


Mean difference is 3933
t_statistic is 1.524180 , p_value is 0.138296


In [30]:
statistic , pvalue = stats.ttest_rel(df2,df1)
print(statistic, pvalue)

1.5241795507101403 0.13829608019283385


In [31]:
p_value/2
# Since we have to check if scheme has raised output or not 

0.06914804009641692

**Answer-3**
Since P-value(=0.069148) is higher than 0.05, we accept NULL hypothesis, and the new scheme has not significantly raised output

**Answer-4**

New Hypotheis Formulation 
> H0 (NULL Hypothesis):- The average output must increse by $5000 i.e. Mean difference MD =(M1-M2)<=5000

>H1 (Alternate Hypothesis):- The average output must increase by $5000 i.e. Mean difference Md(M1-M2)>5000

> Type 1 error rejecting NULL hypothesis when it is true 

> Probability of Type 1 effor is significant level or alpha, which is given as 0.05 or 5%

> NUllHypothesis will be rejected if t_statistic > t_critical

> t_critical = 1.69 for 0.05 significance level and degree of freedom is (sample size-1) i.e.29

In [32]:
t_critical = stats.t.ppf(1-0.05, 29)
print (t_critical)

1.6991270265334972


In [35]:
t_statistic,p_value = ttest_1samp(df2-df1,5000)
print("t_statistic is %f , p_value is %f" %(t_statistic,p_value))

t_statistic is -0.413337 , p_value is 0.682400


In [40]:
D = ((Mean1- Mean2) - 5000) / np.sqrt(((30-1)*np.var(df1) + (30-1)*np.var(df2)) / 30+30 -2)
print (D)

-0.29190642359728486


In [41]:
print(ttest_power(D,alpha=0.05,nobs=30, alternative='smaller'))

0.4667499691950896
